SHA256 hashing in iMessage class

Geoduck7 opened the pull request:
Adding SHA256 hashing to the iMessage class. 

The hash function uses openSSL and creates a hash of the type std::array using the userID, timestamp and message.

Also added chainHash to the iMessage class, but this is still unimplemented.

theis999 commented on src/iMessage.cpp:
```diff
@@ -1,14 +1,37 @@
 #include "iMessage.h"
+#include <openssl/sha.h>
+#include <cstring>
```
Where do you cstring?

Geoduck7 replied:
Used for the SHA256 function

theis999 commented on src/Message.h:
```diff
@@ -6,12 +6,9 @@ class Message {
 
 	Message(std::time_t timestamp, wxString username, wxString messagecontent);
-	
 	wxString FormatToPrint();
```
If you don't have meaningful changes to a file, maybe don't include them.

theis999 commented on src/iMessage.cpp:
```diff
@@ -1,14 +1,37 @@
+iMessage::iMessage(time_t _timestamp, int _member_id, string _text, std::array<std::byte, 32> _hash) :
 	timestamp(_timestamp), member_id(_member_id), text(_text), hash(_hash)
-{ }
+{}
```
Why not `computeHash();`?

theis999 commented on src/iMessage.cpp:
```diff
@@ -1,14 +1,37 @@
+
+iMessage::iMessage(time_t _timestamp, int _member_id, string _text, std::array<std::byte, 32> _hash, std::array<std::byte, 32> _chainHash) :
+	timestamp(_timestamp), member_id(_member_id), text(_text), hash(_hash), chainHash(_chainHash)
+{}
```
Why not `computeHash();`?

Geoduck7 replied:
Added computeHash

theis999 commented on src/iMessage.cpp:
```diff
@@ -1,14 +1,37 @@
+
+	// Create a string combining timestamp, member_id, and text
+	std::ostringstream data_stream;
+	data_stream << timestamp << "|" << member_id << "|" << text;
```
Is there any reason to use a deliminating character between each field?

Geoduck7 replied:
Removed deliminating characters

theis999 reviewed (approved):
Seems alright

Geoduck7 committed:
Added hashing to iMessage

Implemented hashing for iMessage class

Also added chainHash, but still unimplemented.

Changed type of hash to std::array, as the hash always has a fixed size

Removed sha256 from project, as we are using openSSL instead

Geoduck7 committed:
Removed delimination characters in computeHash
